src/axom/sidre/tests/sidre_attribute.cpp

```diff
@@ -56,11 +59,18 @@ const int g_nprotocols = 2;
 const std::string g_protocols[] = {"sidre_json", "json"};
 #endif
 
+const std::set<std::string> g_protocol_saves_attributes {"sidre_hdf5",
+                                                         "sidre_conduit_json",
+                                                         "sidre_json",
+                                                         "sidre_layout_json"};
```

Comment:
Apparently, only some of the Sidre protocols currently save/load attributes.
Not sure if that's by design or desire, but I left it alone for now.

See `saveAttributeLayout` calls in `Group::save()` implementation: 
https://github.com/LLNL/axom/blob/c5a31951a1d71ae574edd2c2ccfa5b42d1216fa2/src/axom/sidre/core/Group.cpp#L1908-L1971